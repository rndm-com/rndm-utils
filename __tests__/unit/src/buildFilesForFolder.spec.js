const fs = require('fs');
const buildFilesForFolder = require('../../../src/buildFilesForFolder');
const readDir = require('../../../src/readDir');
const { describe, test } = global;

describe('src/buildFilesForFolder', () => {
  test('file', () => {
    const base = __dirname;
    const input = {
      files: [
        {
          filename: 'test.text',
          string: 'test',
        }
      ]
    };

    buildFilesForFolder(input, base);

    const file = [__dirname, 'test.text'].join('/');

    const sut = fs.readFileSync(file).toString();
    expect(sut).toMatchSnapshot();

    fs.unlinkSync(file);

  });

  test('files and folders', () => {
    const base = __dirname;
    const input = {
      folders: {
        test: {
          files: [
            {
              filename: 'test.text',
              string: 'test',
            }
          ]
        }
      }
    };

    buildFilesForFolder(input, base);

    const folder = [__dirname, 'test'].join('/');
    const file = [folder, 'test.text'].join('/');

    const sut = fs.readFileSync(file).toString();
    expect(sut).toMatchSnapshot();

    fs.unlinkSync(file);
    fs.rmdirSync(folder);

  });

  test('existing', () => {
    const base = __dirname;

    const file = [base, 'test.text'].join('/');

    fs.writeFileSync(file, 'other');

    const input = {
      files: [
        {
          filename: 'test.text',
          string: 'test',
        }
      ]
    };

    buildFilesForFolder(input, base);

    const sut = fs.readFileSync(file).toString();
    expect(sut).toMatchSnapshot();

    fs.unlinkSync(file);

  });
});
