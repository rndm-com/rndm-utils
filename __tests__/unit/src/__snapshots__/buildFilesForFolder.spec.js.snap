// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/buildFilesForFolder existing 1`] = `"other"`;

exports[`src/buildFilesForFolder file 1`] = `"test"`;

exports[`src/buildFilesForFolder files and folders 1`] = `"test"`;
